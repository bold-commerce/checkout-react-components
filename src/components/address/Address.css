.FieldSet--Address {
  display: flex;
  flex-wrap: wrap;
  flex-direction: column;
}

.FieldSet--Address .stx-select__select-element {
  line-height: initial;
}

.FieldSet--Address .stx-field.Field {
  padding: 0;
}

.FieldSet--Address .stx-input.Field {
    padding: 5px;
    margin: 0;
}

.FieldSet--Address .stx-field__message {
  white-space: nowrap;
}

.FieldSet--Address .stx-input.stx-input--alert {
  margin-bottom: 0.25rem;
}

.FieldSet--Address .stx-field.Field--FirstName,
.FieldSet--Address .stx-field.Field--LastName {
  flex: 1;
}

.Field--Address {
  flex: 2;
}

.Field--Address2 {
  flex: 1;
}

.Field--Company,
.Field--City,
.Field--Country,
.Field--Postal_Code,
.Field--Phone,
.Field--Province {
  flex: 1;
}

@media screen and (max-width: 599px) {
  .FieldSet--Address .FieldGroup {
    flex-direction: column;
    margin: 0;
  }

  .FieldSet--Address .Field {
    margin: 5px;
  }

  .FieldSet--Address .SelectField {
    margin: 5px;
    width: 97%;
  }
}

.stx-field--alert .stx-field__message {
  margin: 0 5px;
  font-style: normal;
  font-weight: normal;
  font-size: 12px;
  line-height: 16px;
}
