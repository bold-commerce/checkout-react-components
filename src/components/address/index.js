// eslint-disable-next-line import/prefer-default-export
export { default as Address } from './Address';
