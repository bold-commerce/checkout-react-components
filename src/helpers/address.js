export const getCountryData = (countryInfo) => {
  const countryData = {};
  countryInfo.forEach((country) => {
    countryData[country.iso_code] = {
      show_province: country.show_province,
      show_postal_code: country.show_postal_code,
      province_label: country.province_label,
    };
  });
  return countryData;
};

export const getCountryList = (countryInfo) => {
  const unsortedCountries = Object.values(countryInfo);
  // eslint-disable-next-line no-nested-ternary
  const countries = unsortedCountries.sort((a, b) => ((a.name > b.name) ? 1 : ((b.name > a.name) ? -1 : 0)));
  return countries;
};

export const verifyAddress = (apiPath, countries, address, csrf) => {
  const selectedCountry = countries.find((country) => country.iso_code === address.country_code);
  const selectedProvince = Object.values(selectedCountry.provinces).find((province) => province.iso_code === address.province_code);
  const errors = {
    first_name: address.first_name.length === 0,
    last_name: address.last_name.length === 0,
    address: address.address_line_1.length === 0,
    city: address.city.length === 0,
    country: address.country_code.length === 0,
    province: address.province_code.length === 0,
    postal_code: false,
    postal_code_error_message: '',
  };

  return fetch(`${apiPath}/validate_address?country_code=${address.country_code}&province=${selectedProvince && selectedProvince.name}&postal_code=${address.postal_code}`, {
    mode: 'cors',
    method: 'GET',
    credentials: 'include',
    headers: {
      'Content-Type': 'application/json',
      'X-CSRF-TOKEN': csrf,
    },
  })
    .then((response) => response.json())
    .then((response) => {
      if (Array.isArray(response.errors)) {
        response.errors.forEach((error) => {
          if (error.field === 'address.postcode') {
            errors.postal_code = true;
          } else {
            errors[error.field] = true;
          }
          if (error.field === 'postal_code' || error.field === 'address.postcode') {
            errors.postal_code_error_message = error.message;
          }
        });

        return errors;
      }

      return null;
    });
};

export const postAddress = (addressEndpoint, countries, address, csrf) => {
  const selectedCountry = countries.find((country) => country.iso_code === address.country_code);
  const selectedProvince = Object.values(selectedCountry.provinces).find((province) => province.iso_code === address.province_code);
  if (!csrf || (selectedCountry && selectedCountry.provinces.length > 0 && !selectedProvince)) return null;

  return fetch(addressEndpoint, {
    mode: 'cors',
    method: 'POST',
    credentials: 'include',
    headers: {
      'Content-Type': 'application/json',
      'X-CSRF-TOKEN': csrf,
    },
    body: JSON.stringify({
      ...address,
      country: selectedCountry?.name ?? null,
      province: selectedProvince?.name ?? null,
    }),
  })
    .then((response) => response.json())
    .then((response) => response.data.application_state)
    .catch(() => null);
};
