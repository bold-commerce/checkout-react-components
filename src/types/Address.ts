export interface Address {
  id: string,
  first_name: string,
  last_name: string,
  address_line_1: string,
  address_line_2: string,
  country: string,
  city: string,
  province: string,
  country_code: string,
  province_code: string,
  postal_code: string,
  business_Name: string,
  phone_number: string 
}