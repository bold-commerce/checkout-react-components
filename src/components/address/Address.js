/* eslint-disable no-mixed-operators */
/* eslint-disable react/forbid-prop-types */
import React from 'react';
import PropTypes from 'prop-types';
import { InputField, SelectField } from '@boldcommerce/stacks-ui';
import './Address.css';

const Address = ({
  address,
  dispatch,
  errors,
  countries,
  provinces,
  showPostalCode,
  showProvince,
  provinceLabel,
  submit,
}) => {
  const countryList = countries.map((countryItem) => <option value={countryItem.iso_code} key={countryItem.iso_code}>{countryItem.name}</option>);
  const provinceList = provinces.map((provinceItem) => <option value={provinceItem.iso_code} key={provinceItem.iso_code}>{provinceItem.name}</option>);

  return (
    <div className="FieldSet--Address">
      <div className="FieldGroup">
        <InputField
          placeholder="First name"
          type="text"
          name="first_name"
          className="Field Field--FirstName"
          value={address.first_name}
          messageType={errors && errors?.first_name && 'alert' || ''}
          messageText={errors && errors?.first_name && 'Enter a first name' || ''}
          onChange={(e) => dispatch({
            type: 'first_name',
            payload: e.target.value,
          })}
          onBlur={submit}
        />
        <InputField
          placeholder="Last name"
          type="text"
          name="last_name"
          className="Field Field--LastName"
          value={address.last_name}
          messageType={errors && errors?.last_name && 'alert' || ''}
          messageText={errors && errors?.last_name && 'Enter a last name' || ''}
          onChange={(e) => dispatch({
            type: 'last_name',
            payload: e.target.value,
          })}
          onBlur={submit}
        />
      </div>
      <div className="FieldGroup">
        <InputField
          placeholder="Company (optional)"
          type="text"
          name="company"
          className="Field Field--Company"
          value={address.business_name}
          onChange={(e) => dispatch({
            type: 'business_name',
            payload: e.target.value,
          })}
          onBlur={submit}
        />
      </div>
      <div className="FieldGroup">
        <InputField
          placeholder="Address"
          type="text"
          name="address"
          className="Field Field--Address"
          value={address.address_line_1}
          messageType={errors && errors?.address && 'alert' || ''}
          messageText={errors && errors?.address && 'Enter an address' || ''}
          onChange={(e) => dispatch({
            type: 'address_line_1',
            payload: e.target.value,
          })}
          onBlur={submit}
        />
        <InputField
          placeholder="Apt, suite, etc."
          type="text"
          name="address2"
          className="Field Field--Address2"
          value={address.address_line_2}
          onChange={(e) => dispatch({
            type: 'address_line_2',
            payload: e.target.value,
          })}
          onBlur={submit}
        />
      </div>
      <div className="FieldGroup">
        <InputField
          placeholder="City"
          type="text"
          name="city"
          value={address.city}
          messageType={errors && errors?.city && 'alert' || ''}
          messageText={errors && errors?.city && 'Enter a city' || ''}
          className="Field Field--City"
          onChange={(e) => dispatch({
            type: 'city',
            payload: e.target.value,
          })}
          onBlur={submit}
        />
      </div>
      <div className="FieldGroup">
        <SelectField
          placeholder="Select a country"
          className="SelectField Field--Country"
          value={address.country_code}
          messageType={errors && errors?.country && 'alert' || ''}
          messageText={errors && errors?.country && 'Select a country' || ''}
          onChange={(e) => dispatch({
            type: 'country_code',
            payload: e.target.value,
          })}
          onBlur={submit}
        >
          {countryList}
        </SelectField>
        {
          address.country_code
            && showProvince && (
            <SelectField
              placeholder={
                `Select a ${provinceLabel}`
              }
              className="SelectField Field--Province"
              value={address.province_code}
              messageType={errors && errors?.province && 'alert' || ''}
              messageText={errors && errors?.province && 'Select a province or state' || ''}
              onChange={(e) => dispatch({
                type: 'province_code',
                payload: e.target.value,
              })}
              onBlur={submit}
            >
              {provinceList}
            </SelectField>
          )
        }
        {address.country_code
          && showPostalCode && (
          <InputField
            placeholder="Postal code / ZIP"
            type="text"
            name="postal"
            className="Field Field--Postal_Code"
            messageType={errors && errors?.postal_code && 'alert' || ''}
            messageText={errors && errors?.postal_code && errors?.postal_code_error_message || ''}
            value={address.postal_code}
            onChange={(e) => dispatch({
              type: 'postal_code',
              payload: e.target.value,
            })}
            onBlur={submit}
          />
        )}
      </div>
      <div className="FieldGroup">
        <InputField
          placeholder="Phone (optional)"
          type="tel"
          name="phone"
          className="Field Field--Phone"
          value={address.phone}
          onChange={(e) => dispatch({
            type: 'phone',
            payload: e.target.value,
          })}
          onBlur={submit}
        />
      </div>
    </div>
  );
};

Address.propTypes = {
  address: PropTypes.any,
  dispatch: PropTypes.func,
  errors: PropTypes.any,
  countries: PropTypes.any,
  provinces: PropTypes.any,
  showPostalCode: PropTypes.bool,
  showProvince: PropTypes.bool,
  provinceLabel: PropTypes.string,
  submit: PropTypes.func,
};

export default Address;
