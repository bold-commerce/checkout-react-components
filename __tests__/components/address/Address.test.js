import React from 'react';
import { render } from '@testing-library/react';
import { Address } from '../../../src/components/address/Address';
import {
  addressErrors, caProvinces, countries, emptyAddress, exampleAddress,
} from '../../utils/addressHelpers';

describe('Address', () => {
  test('renders Address component', () => {
    const { asFragment } = render(
      <Address
        address={emptyAddress}
        onChange={null}
        errors={null}
        countries={countries}
        provinces={caProvinces}
        showPostalCode
        showProvince
        provinceLabel="Province"
        submit={null}
        requiredAddressFields={null}
      />,
    );
    expect(asFragment()).toMatchSnapshot();
  });

  test('renders Address component with errors', () => {
    const { asFragment } = render(
      <Address
        address={emptyAddress}
        onChange={null}
        errors={addressErrors}
        countries={countries}
        provinces={caProvinces}
        showPostalCode
        showProvince
        provinceLabel="Province"
        submit={null}
        requiredAddressFields={null}
      />,
    );
    expect(asFragment()).toMatchSnapshot();
  });

  test('renders Address component with pre-filled data', () => {
    const { asFragment } = render(
      <Address
        address={exampleAddress}
        onChange={null}
        errors={null}
        countries={countries}
        provinces={caProvinces}
        showPostalCode
        showProvince
        provinceLabel="Province"
        submit={null}
        requiredAddressFields={null}
      />,
    );
    expect(asFragment()).toMatchSnapshot();
  });

  test('renders Address component with required fields', () => {
    const requiredFields = [
      'first_name',
      'last_name',
      'address_line_1',
      'address_line_2',
      'business_name',
      'phone_number',
    ];

    const { asFragment } = render(
      <Address
        address={exampleAddress}
        onChange={null}
        errors={null}
        countries={countries}
        provinces={caProvinces}
        showPostalCode
        showProvince
        provinceLabel="Province"
        submit={null}
        requiredAddressFields={requiredFields}
      />,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
