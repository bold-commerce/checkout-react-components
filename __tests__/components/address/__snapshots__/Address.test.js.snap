// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Address renders Address component 1`] = `
<DocumentFragment>
  <div
    class="FieldSet--Address"
  >
    <div
      class="FieldGroup"
    >
      <div
        class="stx-field stx-field--with-input Field Field--FirstName"
      >
        <div
          class="stx-input"
        >
          <input
            class="stx-input__input-element"
            id="stxField1"
            name="first_name"
            placeholder="First name (optional)"
            type="text"
            value=""
          />
        </div>
      </div>
      <div
        class="stx-field stx-field--with-input Field Field--LastName"
      >
        <div
          class="stx-input"
        >
          <input
            class="stx-input__input-element"
            id="stxField2"
            name="last_name"
            placeholder="Last name (optional)"
            type="text"
            value=""
          />
        </div>
      </div>
    </div>
    <div
      class="FieldGroup"
    >
      <div
        class="stx-field stx-field--with-input Field Field--Company"
      >
        <div
          class="stx-input"
        >
          <input
            class="stx-input__input-element"
            id="stxField3"
            name="business_name"
            placeholder="Company (optional)"
            type="text"
            value=""
          />
        </div>
      </div>
    </div>
    <div
      class="FieldGroup"
    >
      <div
        class="stx-field stx-field--with-input Field Field--Address"
      >
        <div
          class="stx-input"
        >
          <input
            class="stx-input__input-element"
            id="stxField4"
            name="address_line_1"
            placeholder="Address (optional)"
            type="text"
            value=""
          />
        </div>
      </div>
      <div
        class="stx-field stx-field--with-input Field Field--Address2"
      >
        <div
          class="stx-input"
        >
          <input
            class="stx-input__input-element"
            id="stxField5"
            name="address_line_2"
            placeholder="Apt, suite, etc."
            type="text"
            value=""
          />
        </div>
      </div>
    </div>
    <div
      class="FieldGroup"
    >
      <div
        class="stx-field stx-field--with-input Field Field--City"
      >
        <div
          class="stx-input"
        >
          <input
            class="stx-input__input-element"
            id="stxField6"
            name="city"
            placeholder="City (optional)"
            type="text"
            value=""
          />
        </div>
      </div>
    </div>
    <div
      class="FieldGroup"
    >
      <div
        class="stx-field stx-field--with-select SelectField Field--Country"
      >
        <div
          class="stx-select stx-select--with-placeholder"
        >
          <select
            class="stx-select__select-element"
            id="stxField7"
          >
            <option
              hidden=""
              value="___PLACEHOLDER___"
            >
              Select a country
            </option>
            <option
              value="CA"
            >
              Canada
            </option>
            <option
              value="US"
            >
              United States
            </option>
          </select>
          <span
            class="stx-select__icon"
          >
            <svg
              class="stx-select-icon__icon"
              viewBox="0 0 24 24"
            >
              <path
                class="stx-select-icon__path"
                d="M3.75261265,6.34153452 C3.38894463,5.92588561 2.75718344,5.88374677 2.34153452,6.24741478 C1.92588561,6.6110828 1.88374677,7.24284399 2.24741478,7.65849291 L11.2524148,18.2364929 C11.6508167,18.6918404 12.3591606,18.6918563 12.757583,18.2365268 L21.763583,7.6585268 C22.1272697,7.24289426 22.0851593,6.61113117 21.6695268,6.24744443 C21.2538943,5.8837577 20.6221312,5.9258681 20.2584444,6.34150064 L12.0050479,16.0594019 L3.75261265,6.34153452 Z"
                fill="#6B6B6B"
              />
            </svg>
          </span>
        </div>
      </div>
    </div>
    <div
      class="FieldGroup"
    >
      <div
        class="stx-field stx-field--with-input Field Field--Phone"
      >
        <div
          class="stx-input"
        >
          <input
            class="stx-input__input-element"
            id="stxField8"
            name="phone_number"
            placeholder="Phone (optional)"
            type="tel"
            value=""
          />
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Address renders Address component with errors 1`] = `
<DocumentFragment>
  <div
    class="FieldSet--Address"
  >
    <div
      class="FieldGroup"
    >
      <div
        class="stx-field stx-field--alert stx-field--with-input Field Field--FirstName"
      >
        <div
          class="stx-input stx-input--alert"
        >
          <input
            class="stx-input__input-element"
            id="stxField9"
            name="first_name"
            placeholder="First name (optional)"
            type="text"
            value=""
          />
        </div>
        <div
          class="stx-field__message"
        >
          Enter a first name
        </div>
      </div>
      <div
        class="stx-field stx-field--alert stx-field--with-input Field Field--LastName"
      >
        <div
          class="stx-input stx-input--alert"
        >
          <input
            class="stx-input__input-element"
            id="stxField10"
            name="last_name"
            placeholder="Last name (optional)"
            type="text"
            value=""
          />
        </div>
        <div
          class="stx-field__message"
        >
          Enter a last name
        </div>
      </div>
    </div>
    <div
      class="FieldGroup"
    >
      <div
        class="stx-field stx-field--alert stx-field--with-input Field Field--Company"
      >
        <div
          class="stx-input stx-input--alert"
        >
          <input
            class="stx-input__input-element"
            id="stxField11"
            name="business_name"
            placeholder="Company (optional)"
            type="text"
            value=""
          />
        </div>
        <div
          class="stx-field__message"
        >
          Enter a company name
        </div>
      </div>
    </div>
    <div
      class="FieldGroup"
    >
      <div
        class="stx-field stx-field--with-input Field Field--Address"
      >
        <div
          class="stx-input"
        >
          <input
            class="stx-input__input-element"
            id="stxField12"
            name="address_line_1"
            placeholder="Address (optional)"
            type="text"
            value=""
          />
        </div>
      </div>
      <div
        class="stx-field stx-field--with-input Field Field--Address2"
      >
        <div
          class="stx-input"
        >
          <input
            class="stx-input__input-element"
            id="stxField13"
            name="address_line_2"
            placeholder="Apt, suite, etc."
            type="text"
            value=""
          />
        </div>
      </div>
    </div>
    <div
      class="FieldGroup"
    >
      <div
        class="stx-field stx-field--with-input Field Field--City"
      >
        <div
          class="stx-input"
        >
          <input
            class="stx-input__input-element"
            id="stxField14"
            name="city"
            placeholder="City (optional)"
            type="text"
            value=""
          />
        </div>
      </div>
    </div>
    <div
      class="FieldGroup"
    >
      <div
        class="stx-field stx-field--alert stx-field--with-select SelectField Field--Country"
      >
        <div
          class="stx-select stx-select--with-placeholder stx-select--alert"
        >
          <select
            class="stx-select__select-element"
            id="stxField15"
          >
            <option
              hidden=""
              value="___PLACEHOLDER___"
            >
              Select a country
            </option>
            <option
              value="CA"
            >
              Canada
            </option>
            <option
              value="US"
            >
              United States
            </option>
          </select>
          <span
            class="stx-select__icon"
          >
            <svg
              class="stx-select-icon__icon"
              viewBox="0 0 24 24"
            >
              <path
                class="stx-select-icon__path"
                d="M3.75261265,6.34153452 C3.38894463,5.92588561 2.75718344,5.88374677 2.34153452,6.24741478 C1.92588561,6.6110828 1.88374677,7.24284399 2.24741478,7.65849291 L11.2524148,18.2364929 C11.6508167,18.6918404 12.3591606,18.6918563 12.757583,18.2365268 L21.763583,7.6585268 C22.1272697,7.24289426 22.0851593,6.61113117 21.6695268,6.24744443 C21.2538943,5.8837577 20.6221312,5.9258681 20.2584444,6.34150064 L12.0050479,16.0594019 L3.75261265,6.34153452 Z"
                fill="#6B6B6B"
              />
            </svg>
          </span>
        </div>
        <div
          class="stx-field__message"
        >
          Select a country
        </div>
      </div>
    </div>
    <div
      class="FieldGroup"
    >
      <div
        class="stx-field stx-field--alert stx-field--with-input Field Field--Phone"
      >
        <div
          class="stx-input stx-input--alert"
        >
          <input
            class="stx-input__input-element"
            id="stxField16"
            name="phone_number"
            placeholder="Phone (optional)"
            type="tel"
            value=""
          />
        </div>
        <div
          class="stx-field__message"
        >
          Enter a phone number
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Address renders Address component with pre-filled data 1`] = `
<DocumentFragment>
  <div
    class="FieldSet--Address"
  >
    <div
      class="FieldGroup"
    >
      <div
        class="stx-field stx-field--with-input Field Field--FirstName"
      >
        <div
          class="stx-input"
        >
          <input
            class="stx-input__input-element"
            id="stxField17"
            name="first_name"
            placeholder="First name (optional)"
            type="text"
            value="John"
          />
        </div>
      </div>
      <div
        class="stx-field stx-field--with-input Field Field--LastName"
      >
        <div
          class="stx-input"
        >
          <input
            class="stx-input__input-element"
            id="stxField18"
            name="last_name"
            placeholder="Last name (optional)"
            type="text"
            value="Doe"
          />
        </div>
      </div>
    </div>
    <div
      class="FieldGroup"
    >
      <div
        class="stx-field stx-field--with-input Field Field--Company"
      >
        <div
          class="stx-input"
        >
          <input
            class="stx-input__input-element"
            id="stxField19"
            name="business_name"
            placeholder="Company (optional)"
            type="text"
            value="Bold Commerce"
          />
        </div>
      </div>
    </div>
    <div
      class="FieldGroup"
    >
      <div
        class="stx-field stx-field--with-input Field Field--Address"
      >
        <div
          class="stx-input"
        >
          <input
            class="stx-input__input-element"
            id="stxField20"
            name="address_line_1"
            placeholder="Address (optional)"
            type="text"
            value="123 Fake Street"
          />
        </div>
      </div>
      <div
        class="stx-field stx-field--with-input Field Field--Address2"
      >
        <div
          class="stx-input"
        >
          <input
            class="stx-input__input-element"
            id="stxField21"
            name="address_line_2"
            placeholder="Apt, suite, etc."
            type="text"
            value="Unit 12"
          />
        </div>
      </div>
    </div>
    <div
      class="FieldGroup"
    >
      <div
        class="stx-field stx-field--with-input Field Field--City"
      >
        <div
          class="stx-input"
        >
          <input
            class="stx-input__input-element"
            id="stxField22"
            name="city"
            placeholder="City (optional)"
            type="text"
            value="Winnipeg"
          />
        </div>
      </div>
    </div>
    <div
      class="FieldGroup"
    >
      <div
        class="stx-field stx-field--with-select SelectField Field--Country"
      >
        <div
          class="stx-select stx-select--with-placeholder"
        >
          <select
            class="stx-select__select-element"
            id="stxField23"
          >
            <option
              hidden=""
              value="___PLACEHOLDER___"
            >
              Select a country
            </option>
            <option
              value="CA"
            >
              Canada
            </option>
            <option
              value="US"
            >
              United States
            </option>
          </select>
          <span
            class="stx-select__icon"
          >
            <svg
              class="stx-select-icon__icon"
              viewBox="0 0 24 24"
            >
              <path
                class="stx-select-icon__path"
                d="M3.75261265,6.34153452 C3.38894463,5.92588561 2.75718344,5.88374677 2.34153452,6.24741478 C1.92588561,6.6110828 1.88374677,7.24284399 2.24741478,7.65849291 L11.2524148,18.2364929 C11.6508167,18.6918404 12.3591606,18.6918563 12.757583,18.2365268 L21.763583,7.6585268 C22.1272697,7.24289426 22.0851593,6.61113117 21.6695268,6.24744443 C21.2538943,5.8837577 20.6221312,5.9258681 20.2584444,6.34150064 L12.0050479,16.0594019 L3.75261265,6.34153452 Z"
                fill="#6B6B6B"
              />
            </svg>
          </span>
        </div>
      </div>
      <div
        class="stx-field stx-field--with-select SelectField Field--Province"
      >
        <div
          class="stx-select stx-select--with-placeholder"
        >
          <select
            class="stx-select__select-element"
            id="stxField24"
          >
            <option
              hidden=""
              value="___PLACEHOLDER___"
            >
              Select a Province
            </option>
            <option
              value="AB"
            >
              Alberta
            </option>
            <option
              value="BC"
            >
              British Columbia
            </option>
            <option
              value="MB"
            >
              Manitoba
            </option>
            <option
              value="NB"
            >
              New Brunswick
            </option>
            <option
              value="NL"
            >
              Newfoundland and Labrador
            </option>
            <option
              value="NT"
            >
              Northwest Territories
            </option>
            <option
              value="NS"
            >
              Nova Scotia
            </option>
            <option
              value="NU"
            >
              Nunavut
            </option>
            <option
              value="ON"
            >
              Ontario
            </option>
            <option
              value="PE"
            >
              Prince Edward Island
            </option>
            <option
              value="QC"
            >
              Quebec
            </option>
            <option
              value="SK"
            >
              Saskatchewan
            </option>
            <option
              value="YT"
            >
              Yukon Territory
            </option>
          </select>
          <span
            class="stx-select__icon"
          >
            <svg
              class="stx-select-icon__icon"
              viewBox="0 0 24 24"
            >
              <path
                class="stx-select-icon__path"
                d="M3.75261265,6.34153452 C3.38894463,5.92588561 2.75718344,5.88374677 2.34153452,6.24741478 C1.92588561,6.6110828 1.88374677,7.24284399 2.24741478,7.65849291 L11.2524148,18.2364929 C11.6508167,18.6918404 12.3591606,18.6918563 12.757583,18.2365268 L21.763583,7.6585268 C22.1272697,7.24289426 22.0851593,6.61113117 21.6695268,6.24744443 C21.2538943,5.8837577 20.6221312,5.9258681 20.2584444,6.34150064 L12.0050479,16.0594019 L3.75261265,6.34153452 Z"
                fill="#6B6B6B"
              />
            </svg>
          </span>
        </div>
      </div>
      <div
        class="stx-field stx-field--with-input Field Field--Postal_Code"
      >
        <div
          class="stx-input"
        >
          <input
            class="stx-input__input-element"
            id="stxField25"
            name="postal"
            placeholder="Postal code / ZIP"
            type="text"
            value="R3A1B3"
          />
        </div>
      </div>
    </div>
    <div
      class="FieldGroup"
    >
      <div
        class="stx-field stx-field--with-input Field Field--Phone"
      >
        <div
          class="stx-input"
        >
          <input
            class="stx-input__input-element"
            id="stxField26"
            name="phone_number"
            placeholder="Phone (optional)"
            type="tel"
            value="204-123-4567"
          />
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Address renders Address component with required fields 1`] = `
<DocumentFragment>
  <div
    class="FieldSet--Address"
  >
    <div
      class="FieldGroup"
    >
      <div
        class="stx-field stx-field--with-input Field Field--FirstName"
      >
        <div
          class="stx-input"
        >
          <input
            class="stx-input__input-element"
            id="stxField27"
            name="first_name"
            placeholder="First name"
            type="text"
            value="John"
          />
        </div>
      </div>
      <div
        class="stx-field stx-field--with-input Field Field--LastName"
      >
        <div
          class="stx-input"
        >
          <input
            class="stx-input__input-element"
            id="stxField28"
            name="last_name"
            placeholder="Last name"
            type="text"
            value="Doe"
          />
        </div>
      </div>
    </div>
    <div
      class="FieldGroup"
    >
      <div
        class="stx-field stx-field--with-input Field Field--Company"
      >
        <div
          class="stx-input"
        >
          <input
            class="stx-input__input-element"
            id="stxField29"
            name="business_name"
            placeholder="Company"
            type="text"
            value="Bold Commerce"
          />
        </div>
      </div>
    </div>
    <div
      class="FieldGroup"
    >
      <div
        class="stx-field stx-field--with-input Field Field--Address"
      >
        <div
          class="stx-input"
        >
          <input
            class="stx-input__input-element"
            id="stxField30"
            name="address_line_1"
            placeholder="Address"
            type="text"
            value="123 Fake Street"
          />
        </div>
      </div>
      <div
        class="stx-field stx-field--with-input Field Field--Address2"
      >
        <div
          class="stx-input"
        >
          <input
            class="stx-input__input-element"
            id="stxField31"
            name="address_line_2"
            placeholder="Apt, suite, etc."
            type="text"
            value="Unit 12"
          />
        </div>
      </div>
    </div>
    <div
      class="FieldGroup"
    >
      <div
        class="stx-field stx-field--with-input Field Field--City"
      >
        <div
          class="stx-input"
        >
          <input
            class="stx-input__input-element"
            id="stxField32"
            name="city"
            placeholder="City (optional)"
            type="text"
            value="Winnipeg"
          />
        </div>
      </div>
    </div>
    <div
      class="FieldGroup"
    >
      <div
        class="stx-field stx-field--with-select SelectField Field--Country"
      >
        <div
          class="stx-select stx-select--with-placeholder"
        >
          <select
            class="stx-select__select-element"
            id="stxField33"
          >
            <option
              hidden=""
              value="___PLACEHOLDER___"
            >
              Select a country
            </option>
            <option
              value="CA"
            >
              Canada
            </option>
            <option
              value="US"
            >
              United States
            </option>
          </select>
          <span
            class="stx-select__icon"
          >
            <svg
              class="stx-select-icon__icon"
              viewBox="0 0 24 24"
            >
              <path
                class="stx-select-icon__path"
                d="M3.75261265,6.34153452 C3.38894463,5.92588561 2.75718344,5.88374677 2.34153452,6.24741478 C1.92588561,6.6110828 1.88374677,7.24284399 2.24741478,7.65849291 L11.2524148,18.2364929 C11.6508167,18.6918404 12.3591606,18.6918563 12.757583,18.2365268 L21.763583,7.6585268 C22.1272697,7.24289426 22.0851593,6.61113117 21.6695268,6.24744443 C21.2538943,5.8837577 20.6221312,5.9258681 20.2584444,6.34150064 L12.0050479,16.0594019 L3.75261265,6.34153452 Z"
                fill="#6B6B6B"
              />
            </svg>
          </span>
        </div>
      </div>
      <div
        class="stx-field stx-field--with-select SelectField Field--Province"
      >
        <div
          class="stx-select stx-select--with-placeholder"
        >
          <select
            class="stx-select__select-element"
            id="stxField34"
          >
            <option
              hidden=""
              value="___PLACEHOLDER___"
            >
              Select a Province
            </option>
            <option
              value="AB"
            >
              Alberta
            </option>
            <option
              value="BC"
            >
              British Columbia
            </option>
            <option
              value="MB"
            >
              Manitoba
            </option>
            <option
              value="NB"
            >
              New Brunswick
            </option>
            <option
              value="NL"
            >
              Newfoundland and Labrador
            </option>
            <option
              value="NT"
            >
              Northwest Territories
            </option>
            <option
              value="NS"
            >
              Nova Scotia
            </option>
            <option
              value="NU"
            >
              Nunavut
            </option>
            <option
              value="ON"
            >
              Ontario
            </option>
            <option
              value="PE"
            >
              Prince Edward Island
            </option>
            <option
              value="QC"
            >
              Quebec
            </option>
            <option
              value="SK"
            >
              Saskatchewan
            </option>
            <option
              value="YT"
            >
              Yukon Territory
            </option>
          </select>
          <span
            class="stx-select__icon"
          >
            <svg
              class="stx-select-icon__icon"
              viewBox="0 0 24 24"
            >
              <path
                class="stx-select-icon__path"
                d="M3.75261265,6.34153452 C3.38894463,5.92588561 2.75718344,5.88374677 2.34153452,6.24741478 C1.92588561,6.6110828 1.88374677,7.24284399 2.24741478,7.65849291 L11.2524148,18.2364929 C11.6508167,18.6918404 12.3591606,18.6918563 12.757583,18.2365268 L21.763583,7.6585268 C22.1272697,7.24289426 22.0851593,6.61113117 21.6695268,6.24744443 C21.2538943,5.8837577 20.6221312,5.9258681 20.2584444,6.34150064 L12.0050479,16.0594019 L3.75261265,6.34153452 Z"
                fill="#6B6B6B"
              />
            </svg>
          </span>
        </div>
      </div>
      <div
        class="stx-field stx-field--with-input Field Field--Postal_Code"
      >
        <div
          class="stx-input"
        >
          <input
            class="stx-input__input-element"
            id="stxField35"
            name="postal"
            placeholder="Postal code / ZIP"
            type="text"
            value="R3A1B3"
          />
        </div>
      </div>
    </div>
    <div
      class="FieldGroup"
    >
      <div
        class="stx-field stx-field--with-input Field Field--Phone"
      >
        <div
          class="stx-input"
        >
          <input
            class="stx-input__input-element"
            id="stxField36"
            name="phone_number"
            placeholder="Phone"
            type="tel"
            value="204-123-4567"
          />
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
